modules/programs/pandoc.nix

```
@@ -0,0 +1,107 @@
+{ config, lib, pkgs, ... }:
+
+let
+
+  cfg = config.programs.pandoc;
+
+  inherit (lib) literalExpression mkEnableOption mkIf mkOption types;
+
+  defaultsFormat = pkgs.formats.json { };
+
+  makeTemplateFile = name: file:
+    lib.nameValuePair "pandoc/templates/${name}" { source = file; };
+
+  getFileName = file:
+    # This is actually safe here, since it is just a file name
+    builtins.unsafeDiscardStringContext (baseNameOf file);
+
+  makeCslFile = file:
+    lib.nameValuePair "pandoc/csl/${getFileName file}" { source = file; };
+
+in rec {
+  meta.maintainers = [ lib.maintainers.kirelagin ];
+
+  options.programs.pandoc = {
+    enable = mkEnableOption "pandoc";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pandoc;
+      defaultText = literalExpression "pkgs.pandoc";
+      description = "The pandoc package to use.";
+    };
+
+    # We wrap the executable to pass some arguments
+    wrappedPackage = mkOption {
+      readOnly = true;
+      type = types.package;
+      description = "Resulting package";
```

Descriptions should be proper sentences, i.e., end with a period.
```suggestion
      description = "Resulting package.";
```
Same comment holds for some of the other options.